packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx

```diff
@@ -3,45 +3,56 @@ import { v4 } from "uuid"
 import classnames from "classnames"
 import { history } from "prosemirror-history"
 import { keymap } from "prosemirror-keymap"
-import { EditorState } from "prosemirror-state"
 import { Node, Schema } from "prosemirror-model"
 import { Label } from "@kaizen/draft-form"
 import { baseKeymap } from "prosemirror-commands"
-import { useRichTextEditor } from "@cultureamp/rich-text-toolkit"
+import {
+  useRichTextEditor,
+  markIsActive,
+  removeMark,
+  addMark,
+} from "@cultureamp/rich-text-toolkit"
 import { OverrideClassName } from "@kaizen/component-base"
 import { EditorContentArray } from "./types"
 import { createSchemaFromControls } from "./schema"
+import { createInitialState } from "./state"
 import { buildKeymap } from "./keymap"
 import { buildInputRules } from "./inputrules"
 import styles from "./RichTextEditor.scss"
+import { Toolbar, ToolbarSection, ToggleIconButton } from "./"
 
+type ToolbarControls = "strong" | "em" | "underline"
 export interface RichTextEditorProps
   extends OverrideClassName<Omit<HTMLAttributes<HTMLDivElement>, "onChange">> {
   onChange: (content: EditorContentArray) => void
   value: EditorContentArray
   labelText: ReactNode
+  controls?: ToolbarControls[][]
 }
 
 export const RichTextEditor: React.VFC<RichTextEditorProps> = props => {
-  const { onChange, value, labelText, classNameOverride, ...restProps } = props
-  const [schema] = useState<Schema>(createSchemaFromControls([]))
+  const {
+    onChange,
+    value,
+    labelText,
+    classNameOverride,
+    controls,
+    ...restProps
+  } = props
+  const [schema] = useState<Schema>(
+    createSchemaFromControls(controls?.flat() || [])
+  )
+  const marksFromControls = controls?.map(controlGroup =>
+    controlGroup.map(control => schema.marks[control])
+  )
   const [labelId] = useState<string>(v4())
-  const [editorRef, editorState] = useRichTextEditor(
-    EditorState.create({
-      doc: value
-        ? Node.fromJSON(schema, {
-            type: "doc",
-            contentObject: value,
-          })
-        : null,
-      schema,
-      plugins: [
-        history(),
-        keymap(buildKeymap(schema)),
-        keymap(baseKeymap),
-        buildInputRules(schema),
-      ],
-    }),
+  const [editorRef, editorState, dispatchTransaction] = useRichTextEditor(
+    createInitialState(value, schema, [
```

Comment:
I had inlined this stuff so that we didn't need that extra `state.ts` file. Did you intend to change this back to the old way?

Reply:
True its not entirely necessary, my thinking was to have it export more like a helper function so it could be more easily digestible to a consumer / someone referencing this if they need to build a custom RTE.